plugins/services/src/js/containers/services/ServiceTreeView.js

```diff
@@ -1,94 +1,98 @@
-import ServiceSidebarFilters from './ServiceSidebarFilters';
-import ServiceTree from '../../structs/ServiceTree';
+import Service from '../../structs/Service';
 import ServicesTable from './ServicesTable';
 
-import Breadcrumbs from '../../../../../../src/js/components/Breadcrumbs';
-import FilterBar from '../../../../../../src/js/components/FilterBar';
-import FilterHeadline from '../../../../../../src/js/components/FilterHeadline';
+import DSLFilterList from '../../../../../../src/js/structs/DSLFilterList';
+import DSLFilterField from '../../../../../../src/js/components/DSLFilterField';
+import DSLExpression from '../../../../../../src/js/structs/DSLExpression';
 
 class ServiceTreeView extends React.Component {
+  getButtonBar() {
+    const {modalHandlers} = this.context;
+    const {filterExpression} = this.props;
 
-  getHeadline() {
-    const {services} = this.props;
-
-    if (services.filters.searchString) {
-      return (
-        <ul className="breadcrumb-style-headline list-unstyled list-inline">
-          <li className="h4">
-            Showing results for "{services.filters.searchString}"
-          </li>
-          <li className="h4 clickable" onClick={this.props.clearFilters}>
-            <a className="small">
-              (Clear)
-            </a>
-          </li>
-        </ul>
-      );
+    if (filterExpression.defined) {
+      return null;
     }
 
-    if (Object.keys(services.filters).length) {
-      return (
-        <FilterHeadline
-          className="breadcrumb-style-headline"
-          onReset={this.props.clearFilters}
-          name="Service"
-          currentLength={services.filtered.length}
-          totalLength={services.all.length} />
-      );
-    }
+    return (
+      <div className="column-medium-7 text-align-right">
+        <button className="button button-stroke"
+          onClick={modalHandlers.createGroup}>
+          Create Group
+        </button>
+        {' '}
+        <button className="button button-success"
+          onClick={modalHandlers.createService}>
+          Run a Service
+        </button>
+      </div>
+    );
+  }
+
+  getFilterBar() {
+    const {
+      filters,
+      filterExpression,
+      onFilterExpressionChange
+    } = this.props;
+
+    let hostClasses = classNames({
+      'column-medium-5': !filterExpression.value,
+      'column-medium-12': filterExpression.value
+    });
 
     return (
-      <Breadcrumbs routes={this.props.routes} params={this.props.params} />
+      <div className="row">
+        <div className={hostClasses}>
+          <DSLFilterField
+            filters={filters}
+            expression={filterExpression}
+            onChange={onFilterExpressionChange} />
+        </div>
+        {this.getButtonBar()}
+      </div>
     );
   }
 
+  getSearchHeader() {
+    let {filterExpression} = this.props;
+    if (filterExpression.defined) {
+      return (
+        <h5 className="muted">Search Results</h5>
```

Comment:
True, but I would say that the pain of rebasing the chain of PRs is bigger than the size of this right now.